shell/browser/ui/file_dialog_linux.cc

```diff
@@ -0,0 +1,261 @@
+// Copyright (c) 2024 Microsoft, Gmbh.
+// Use of this source code is governed by the MIT license that can be
+// found in the LICENSE file.
+
+#include <memory>
+#include <string>
+
+#include "base/files/file_util.h"
+#include "base/functional/bind.h"
+#include "base/functional/callback.h"
+#include "base/memory/raw_ptr.h"
+#include "base/memory/raw_ptr_exclusion.h"
+#include "base/run_loop.h"
+#include "base/strings/string_util.h"
+#include "base/strings/utf_string_conversions.h"
+#include "shell/browser/javascript_environment.h"
+#include "shell/browser/native_window_views.h"
+#include "shell/browser/ui/file_dialog.h"
+#include "shell/browser/ui/select_file_policy.h"
+#include "shell/common/gin_converters/callback_converter.h"
+#include "shell/common/gin_converters/file_path_converter.h"
+#include "ui/gtk/select_file_dialog_linux_gtk.h"  // nogncheck
+#include "ui/shell_dialogs/select_file_dialog.h"
+#include "ui/shell_dialogs/selected_file_info.h"
+
+namespace file_dialog {
+
+DialogSettings::DialogSettings() = default;
+DialogSettings::DialogSettings(const DialogSettings&) = default;
+DialogSettings::~DialogSettings() = default;
+
+namespace {
+
+ui::SelectFileDialog::Type GetDialogType(int properties) {
+  if (properties & OPEN_DIALOG_OPEN_FILE)
+    return ui::SelectFileDialog::SELECT_OPEN_FILE;
+
+  if (properties & OPEN_DIALOG_OPEN_DIRECTORY)
+    return ui::SelectFileDialog::SELECT_FOLDER;
+
+  if (properties & OPEN_DIALOG_MULTI_SELECTIONS)
```

Comment:
The earlier behavior was to use `GTK_FILE_CHOOSER_ACTION_SELECT_FOLDER` for directories and `GTK_FILE_CHOOSER_ACTION_OPEN` for files and setting `gtk_file_chooser_set_select_multiple` if multi-selections bit was set.

`SELECT_OPEN_MULTI_FILE` would result in a call to `CreateMultiFileOpenDialog()` which always uses `GTK_FILE_CHOOSER_ACTION_OPEN`. Can multiple directories be selected with this?

Reply:
ah - i see the issue. I'll push a fix momentarily!

Reply:
done in 255f3bf0e0